pkg/util/ranger/ranger.go

```diff
@@ -735,6 +735,9 @@ func points2EqOrInCond(ctx expression.BuildContext, points []*point, col *expres
 	retType := col.GetType(ctx.GetEvalCtx())
 	args := make([]expression.Expression, 0, len(points)/2)
 	args = append(args, col)
+	if len(points) == 2 && points[0].value.IsNull() && points[0].value.IsNull() {
+		return expression.NewFunctionInternal(ctx, ast.IsNull, retType, args...)
+	}
```

Comment:
It should be possible that there are multiple values and one of them is `NULL`.
We need to consider the case `isnull or eq/in

Reply:
I have refactored this code.